### PR TITLE
[Plugin] Refactor add_blockdev_cmd into add_device_cmd

### PR DESCRIPTION
**Pull request by TurboTurtle**
Initially, the design behind `add_blockdev_cmd()` made the assumption
that there would (eventually) be separate methods for different kinds of
devices. On review, this is less practical and less maintainable than a
single method, `add_device_cmd()` paired with improved functionality for
handling device lists.

As such, rework `add_blockdev_cmd()` into `add_device_cmd()` and with
it, change the logic on how device lists are sourced. The `devices` dict
handed to plugins now has a top-level separation between device types,
in this revision putting `block` and `fibre` devices under `storage` in
the `devices` dict.

In using `add_device_cmd()`, plugins may now specify either `block`,
`fibre`, or `storage` for both. In addition, any devices passed via the
`devices` parameter that aren't keys in the devices dict are returned,
meaning a plugin may specify a list of devices and add `storage` to
include all storage devices for command iteration.

This change should allow easier addition of network device iteration.

Signed-off-by: Jake Hunsaker <jhunsake@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by TurboTurtle** [comment]
Note, once #2972 is merged I'll need to update this one to include loop devices. The only question in my mind is do we put loop devices into the block device list always, or keep them separated in a (new) `devices['storage']['loop']` element?

**Comment by jcastill** [comment]
> Note, once #2972 is merged I'll need to update this one to include loop devices. The only question in my mind is do we put loop devices into the block device list always, or keep them separated in a (new) `devices['storage']['loop']` element?

I don't think we should separate them, since loop devices are block devices, And I cannot think what command (other than losetup) would need to be run that other block devs cannot.

**Review by pmoravec — APPROVED** [review]
A minor nitpick: we shall update https://github.com/sosreport/sos/blob/main/plugins_overview.py#L15 :)

**Comment by TurboTurtle** [comment]
Rebased to include the loop device fix, waiting for remaining CI jobs to finish.

File: sos/report/__init__.py
Comment (pmoravec) [inline sos/report/__init__.py:504]: Nice improvement :)

File: sos/report/plugins/scsi.py
Comment (pmoravec) [inline sos/report/plugins/scsi.py:64]: I dont have means to test this, but I believe the `prepend_path` option is really not further needed.

**Review by pmoravec — APPROVED** [review]

File: sos/report/plugins/scsi.py
Comment (TurboTurtle) [reply sos/report/plugins/scsi.py:64]: At the moment, correct. I'm a bit hesitant to remove it because I think there may be a use for it in the future, potentially (e.g. a container-focused sos report that may need some host-level collections as well), but I don't know yet how practical that actually is or if it's entirely irrelevant.

I'd rather leave it for now and remove it after awhile of no usage than pulling it out now and having to replace it later.

**Pull request merged**
